explorer/src/Pos/Explorer/Web/Api.hs

```diff
@@ -18,85 +19,104 @@ module Pos.Explorer.Web.Api
 
 import           Universum
 
+import           Control.Monad.Catch          (try, tryJust)
 import           Data.Proxy                   (Proxy (Proxy))
 
 import           Pos.Explorer.Web.ClientTypes (Byte, CAda, CAddress, CAddressesFilter,
                                                CAddressSummary, CBlockEntry, CBlockSummary,
                                                CGenesisAddressInfo, CGenesisSummary,
                                                CHash, CTxBrief, CTxEntry, CTxId,
                                                CTxSummary)
-import           Pos.Explorer.Web.Error       (ExplorerError)
+import           Pos.Explorer.Web.Error       (ExplorerError, _Internal)
 import           Pos.Types                    (EpochIndex)
+import           Pos.Util.Servant             (ModifiesApiRes (..), VerbMod)
 import           Servant.API                  ((:<|>), (:>), Capture, Get, JSON,
                                                QueryParam)
 
-
 type PageNumber = Integer
 
+-- | API result modification mode used here.
+data ExplorerVerbTag
+
+-- | Wrapper for Servants 'Verb' data type,
+-- which allows to catch exceptions thrown by Explorer's endpoints.
+type ExplorerVerb verb = VerbMod ExplorerVerbTag verb
+
+-- | Shortcut for common api result types.
+type ExRes verbMethod a = ExplorerVerb (verbMethod '[JSON] a)
+
+instance ModifiesApiRes ExplorerVerbTag where
+    type ApiModifiedRes ExplorerVerbTag a = Either ExplorerError a
+    modifyApiResult _ =
+        try . catchEndpointErrors . (either throwM pure =<<)
+        where
+            catchEndpointErrors :: (MonadCatch m) => m a -> m (Either ExplorerError a)
+            catchEndpointErrors = tryJust $ \e -> (e ^? _Internal) $> e
```

Comment:
Here there is no need to do so since you have only one constructor of `ExplorerError`
Just `try` should suffice